h2o-genmodel/src/main/java/hex/genmodel/easy/OneHotEncoderColumnMapper.java

```diff
@@ -18,11 +18,20 @@ public OneHotEncoderColumnMapper(GenModel m) {
     String[][] origDomainValues = _m.getOrigDomainValues();
     Map<String, Integer> columnMapping = new HashMap<>(origNames.length);
     int pos = 0;
+    // non-categorical
+    for (int i = 0; i < _m.getOrigNumCols(); i++) {
+      if (origDomainValues[i] != null)
+        continue;
+      columnMapping.put(origNames[i], pos);
+      pos++;
+    }
+    // categorical
     for (int i = 0; i < _m.getOrigNumCols(); i++) {
-      String colName = origNames[i];
-      columnMapping.put(colName, pos);
       String[] domainValues = origDomainValues[i];
-      pos += domainValues != null ? domainValues.length + 1 : 1; 
+      if (domainValues == null)
+        continue;
+      columnMapping.put(origNames[i], pos);
+      pos += domainValues.length + 1;
```

Comment:
Why is `+1` needed ? Advancing for `domainValues.length` should be enough for the reader ?

Reply:
The last one is for NA (missing).

Reply:
I see. Thanks.